doc/dependency-resolution.md

```diff
@@ -1,356 +1,129 @@
+        "content_id": "03531ffd-314b-4fed-9361-fa9816005c31",
+        "description": "Apply for a driving licence, tax your vehicle, book your driving test, and find out the legal requirements for buying, owning, importing or scrapping a car or motorcycle, and read about your rights and responsibilities as a driver",
+        "document_type": "mainstream_browse_page",
+        "locale": "en",
+        "public_updated_at": "2016-10-07T22:18:32Z",
+        "schema_name": "mainstream_browse_page",
+        "title": "Driving and transport",
+        "links": {},
+        "api_url": "https://www.gov.uk/api/content/browse/driving",
+        "web_url": "https://www.gov.uk/browse/driving"
+      }
+    ],
+  }
+}
+```
+@TODO - get a better example with hierarchy and translations
 
-Once it has visibility over the dependency rules for each of the groups, it
-should run a query against the database to detect content items that need to be
-updated as a result of the incoming request. It will do this using Postgres's
-WITH RECURSIVE functionality. This was explored in the following branch to
-demonstrate that it is feasible to detect dependent nodes in this way:
+As the Content Store holds a static representation of a content item - including
+it’s links - any changes to the content item or any of it’s links requires
+re-representing the item in the Content Store. This can mean that for a single
+update of a content item the Content Store receives updates of many items it
+stores.
 
-[https://github.com/alphagov/publishing-api/tree/dependency-transitive-closure](https://github.com/alphagov/publishing-api/tree/dependency-transitive-closure)
+We consider these links to between content items to be dependencies between
+content items.
 
-We should make it so that this query only follows edges in the dependency graph
-for groups which contain any of the fields that have changed on the content
-item. The branch linked to above demonstrates that it is possible to restrict
-edges followed by a link_type (which is another name for 'group'). The recursive
-behaviour of this query depends on the boolean configuration setting for that
-group and should be factored into the query.
+## Determining the links to show with a content item
 
-All of this behaviour should be completed out-of-band from the publishing
-application's request cycle with the Publishing API. It is proposed that we
-isolate all of this dependency resolution work into a separate Sidekiq worker
-that is responsible for determining the set of content items that need to be
-updated and for actually presenting and sending those updates downstream. When
-the Publishing API receives any interaction that can affect dependencies, it
-need only place this content item on a 'dirty' queue for processing by the
-dependency resolution worker.
+At the point of representing a content item to a Content Store we need to load
+the details of every content item that has an explicit or implicit link from
```

Comment:
I'm not sure I've heard of these transitive links, or dependencies referred to as "implicit" before. This is not how I think about them, as because of the way that link expansion can be configured, there may be no meaningful relationship between items that have some transitive dependency,